key.go

```diff
@@ -0,0 +1,69 @@
+package statter
+
+import (
+	"sync"
+	"unsafe"
+)
+
+var keyPool = sync.Pool{
+	New: func() interface{} {
+		return &key{b: make([]byte, 0, 256)}
+	},
+}
+
+type key struct {
+	b []byte
+}
+
+func newKey(name string, tags []Tag) *key {
+	k := keyPool.Get().(*key)
+	k.b = k.b[:0]
+
+	k.b = append(k.b, name...)
+
+	// Short path for no tags.
+	if len(tags) == 0 {
+		return k
+	}
+
+	// The tags must be sorted to create a consistent key.
+	sortTags(tags)
+
+	for _, tag := range tags {
+		k.b = append(k.b, ':')
+		k.b = append(k.b, tag[0]...)
+		k.b = append(k.b, '=')
+		k.b = append(k.b, tag[1]...)
+	}
+
+	return k
+}
+
+// String returns the key as a string.
+//
+// The returned string should only be used while
+// holding a reference to the key.
+func (k *key) String() string {
+	return *(*string)(unsafe.Pointer(&k.b))
+}
```

Comment:
Why the unsafe and pointer shenanigans here? Is it for performance? I imagine that you can't just `fmt.Sprintf` the []byte without losing a huuuuge amount of perf but surely there must be another way, no? It looks pretty spooky from my uneducated eyes when it comes to perf optimizations haha.

Reply:
All this does is convert a byte slice to a string using the exact underlying array. Where I to `string(k.b)` I would get an alloc, and as I only used the string in the key while I hold it, it is cheaper to do it this way and avoid the alloc. This is also how `strings.Builder` achieves this effect: https://golang.org/src/strings/builder.go?s=1395:1428#L37.